extensions/gdx-box2d/gdx-box2d/src/com/badlogic/gdx/physics/box2d/AABBUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.badlogic.gdx.physics.box2d;
+
+import com.badlogic.gdx.math.Vector2;
+
+public class AABBUtil {
+    /** Sets the lower and upper bounds of the specified fixture to the provided Vector2 values.
+     * Supported Shapes: Polygon/Circle **/
+    public static void getAABB(Fixture fixture, Vector2 lowerBound, Vector2 upperBound) {
+        Body body = fixture.getBody();
+        Transform transform = body.getTransform();
+
+        Shape shape = fixture.getShape();
+        switch (shape.getType()) {
+            case Polygon:
+                PolygonShape polygon = (PolygonShape) shape;
+                getPolygonAABB(polygon, transform, lowerBound, upperBound);
+                break;
+            case Circle:
+                CircleShape circle = (CircleShape) shape;
+                getCircleAABB(circle, transform, lowerBound, upperBound);
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported shape type: " + shape.getType());
+        }
+    }
+
+    private static void getPolygonAABB(PolygonShape polygon, Transform transform, Vector2 lowerBound, Vector2 upperBound) {
+        Vector2 vertex = new Vector2();
+        lowerBound.set(Float.MAX_VALUE, Float.MAX_VALUE);
+        upperBound.set(-Float.MAX_VALUE, -Float.MAX_VALUE);
+
+        for (int i = 0; i < polygon.getVertexCount(); i++) {
+            polygon.getVertex(i, vertex);
+            transform.mul(vertex);
```

Comment:
Like this?

Reply:
```Vector2 transformedVertex = transform.mul(vertex);

        for (int i = 0; i < polygon.getVertexCount(); i++) {
            polygon.getVertex(i, vertex);

            lowerBound.x = Math.min(lowerBound.x, transformedVertex.x);
            lowerBound.y = Math.min(lowerBound.y, transformedVertex.y);
            upperBound.x = Math.max(upperBound.x, transformedVertex.x);
            upperBound.y = Math.max(upperBound.y, transformedVertex.y);
        }```

Reply:
https://github.com/libgdx/libgdx/pull/7538/commits/5c969c803a842cbdeb5be0ff6dab5320991b920a

Reply:
Are you sure? I think every vertex has to be transformed

Reply:
I think what @jose-puente meant is that you can calculate the min and max corners of the axis-aligned bounding box, and then you just need to transform the min corner and max corner. That wouldn't be true though, if Transform can rotate the polygon, since that changes the AABB. I think transforming every vertex is correct if the AABB is meant to be calculated from the transformed Polygon. Transforming only the min and max AABB vertices might be correct, big maybe, if the Transform is meant to apply to the AABB (making it no longer always axis-aligned). So, I think this code was probably right in its original form...

Reply:
Yes, my fault. I didn't consider the effect of a rotation